### ISSUE TITLE
Performance issues while holding down a key

### ISSUE THREAD

**Issue by Spatbee** [lightningboltemoji/clear-unsent-messages#2]
when holding down a key to skip/select dialogue performance is severely affected

I took a look through the code and the most likely culprit is the timers being created
I would suggest not making a new timer if a timer was created recently as a fix

**Comment by lightningboltemoji** [comment]
Thanks for reporting! 

I tried to recreate this by talking to an NPC and holding down space and holding down 1 for the first chat option. My FPS didn't change.

Could you tell me more about where this is happening/how I should recreate?

**Comment by Spatbee** [comment]
I hold down the button *before* interacting with the boats on fossil island, for example.  I can reproduce the drop in framerate by holding down any button outside of dialogue for about a second or longer.  It takes my framerate from 50 (capped) down to about 22.

I have my "repeat rate" on my keyboard set to max, when I tried lowering it the problem went away.  So if you can't reproduce the problem that is probably why.

**Comment by lightningboltemoji** [comment]
Sorry for the long wait.

I bumped my repeat rate as high as it could go on macOS with `defaults write -g KeyRepeat -int 1` and I might _kinda_ be able to reproduce haha. I do see lows of 90 FPS holding a key vs 100-110 at idle. But it doesn't matter much -- appreciate you testing different repeat rates/confirming that theory.

I took your suggestion of "do less": fe2dc6fdc3d2caa547f2ca13584c98e6ef165109 In my pseudo-scientific benchmark, the part that runs on the client thread is now ~7x faster, which where slowness would hurt rendering. Hopefully it makes a difference for you!
